src/test/java/SplunkDockerImageTest.java

```diff
@@ -0,0 +1,61 @@
+import com.splunk.Service;
+import com.splunk.ServiceArgs;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import java.util.Arrays;
+
+public class SplunkDockerImageTest {
+
+    /**
+     * Connects to the specified Splunk server.
+     * @param dockerImage Docker image to retrieve all connection parameters from
+     */
+    private void connectToSplunk(DockerImage dockerImage) {
+        String password = Arrays.stream(dockerImage.getEnvironmentVariables())
+                .filter(environmentVariable -> environmentVariable.startsWith("SPLUNK_PASSWORD="))
+                .map(environmentVariable -> environmentVariable.substring(16))
+                .findFirst()
+                .orElse("");
+
+        // Create a map of arguments and add login parameters
+        ServiceArgs loginArgs = new ServiceArgs();
+        loginArgs.setUsername("admin");
+        loginArgs.setPassword(password);
+        loginArgs.setHost("localhost");
+        loginArgs.setPort(8089);
+
+        System.setProperty("https.protocols", "TLSv1,TLSv1.1,TLSv1.2");
+
+        // Create a Service instance and log in with the argument map
+        Service service = Service.connect(loginArgs);
+
+        System.out.println(service.getApplications());
+    }
+
+    @ParameterizedTest
+    @CsvSource(
+            {
+                    "SPLUNK_LATEST"
+            }
+    )
+    public void testSplunkDockerImage(DockerImage dockerImage) throws InterruptedException {
+        DockerUtility dockerUtility = new DockerUtility();
+        String containerName = "splunk_container";
+        try {
+            // Create and run container
+            dockerUtility.createAndRunContainer(dockerImage, containerName, true);
+
+            // Wait 20s for the Splunk server to start up
+            Thread.sleep(20_000);
+
+            // Test if the connection to the database is successful
+            Assertions.assertDoesNotThrow(() -> connectToSplunk(dockerImage));
+        }
```

Comment:
is there a possibility of exception being thrown - do we need a catch block here, with a failed assertion?
Can we format finally to be in the same line as the finishing brace of try?

Reply:
1. Yes, there's a possibility of a RuntimeException being thrown when the docker container is created and run. Irrespective of whether an exception is thrown or not, we need to clean the container if it exists, hence the placement in the finally block.
2. I've brought the finally line to the same line as the ending brace for try.

Reply:
Discussed offline, and we're essentially making a distinction between test env vs application correctness.